R/utilities.R

```diff
@@ -3146,6 +3153,116 @@ get_bams = function(this_sample_id,
   return(bam_details)
 }
 
+#' @title View a variant in IGV
+#'
+#' @description Load bam(s) and view the context around a mutation
+#'
+#' @details Load bam(s) and view the context around a mutation
+#'
+#' @param this_mutation Specify the mutation of interest in MAF format.
+#' @param igv_port Specify the port IGV is listening on.
+#' @param socket 
+#' @param sort_by base, quality, sample or readGroup
+#'
+#' @return Path to file (.png).
+#'
+#' @import SRAdb
+#' @export
+#'
+#' @examples
+#' #IMPORTANT: you must be running IGV on the host that is running R and you need to have it listening on a port
+#' # The simplest scenario is to run this command on a terminal (if using a Mac), assuming you are using R on gphost10 and you have a ssh config that routes gp10 to that host
+#' # ssh -X gp10
+#' # then launch IGV (e.e. from a conda installation):
+#' # conda activate igv; igv &
+#' # socket = make_igv_snapshot() #run with no arguments to get the socket for a running IGV instance
+#' # this_mutation = get_coding_ssm(seq_type="capture") %>% head(1)
+#' # view_mutation_igv(this_mutation,socket=socket,this_seq_type="capture",colour_by="READ_STRAND",squish=TRUE,viewaspairs=TRUE)
+view_mutation_igv = function(
+                             this_mutation,
+                             this_seq_type="genome",
+                             igv_port = 60506,
+                             socket,
+                             sort_by="base",
+                             colour_by,
+                             squish=FALSE,
+                             viewaspairs=FALSE){
+  if(missing(socket)){
+    print("returning socket for future use")
+    sock = IGVsocket(port = igv_port)
+    if(exists("currently_loaded_bam")){
+      currently_loaded_bam <<- "" #avoid the function mistakenly thinking IGV still has a bam loaded
+    }
+    return(sock)
+  }
+  this_sample_id = unique(this_mutation$Tumor_Sample_Barcode)
+  if(length(this_sample_id)>1){
+    stop("provide a MAF with only one sample_id")
+  }
+  
+  start = pull(this_mutation,Start_Position)
+  end = start
+  chrom = pull(this_mutation,Chromosome)
+  region = paste0(chrom,":",start-50,"-",end+50)
+  sock = socket
+  
+    meta = get_gambl_metadata(seq_type_filter=this_seq_type) %>% 
+      dplyr::filter(sample_id %in% this_sample_id)
+    
+      genome_build = pull(meta,genome_build)
+    
+    bam_path_pattern = "/projects/rmorin/projects/gambl-repos/gambl-rmorin/data/{seq_type}_bams/{sample_id}.{genome_build}.bam"
```

Comment:
Maybe this piece here can be swapped to use the `get_bams()` output? But this can be done in another PR with small improvements